mypy/checker.py

```diff
@@ -2424,6 +2427,31 @@ def is_literal_none(n: Node) -> bool:
     return isinstance(n, NameExpr) and n.fullname == 'builtins.None'
 
 
+def and_conditional_maps(m1: Optional[TypeMap], m2: Optional[TypeMap]) -> Optional[TypeMap]:
+    if m1 is None or m2 is None:
+        # One of the branches can never occur.
+        return None
+    # Both branches are possible; combine the information.
```

Comment:
Add a mention that the `m2` argument takes precedence if the same node happens to be in both maps. I wonder if a more correct behavior would be to generate an intersection type? Would as we don't have intersection types, the current behavior is fine (and it's the same as it used to be).